shared/ui-components/src/helpers/useKeyPress.ts

```diff
@@ -0,0 +1,37 @@
+import { useState, useEffect, useCallback } from "react"
+
+function useKeyPress(targetKey: string) {
+  const [keyPressed, setKeyPressed] = useState<boolean>(false)
+
+  const downHandler = useCallback(
+    ({ key }: KeyboardEvent) => {
+      if (key === targetKey) {
+        setKeyPressed(true)
+      }
+    },
+    [targetKey]
+  )
+
+  const upHandler = useCallback(
+    ({ key }: KeyboardEvent) => {
+      if (key === targetKey) {
+        setKeyPressed(false)
+      }
+    },
+    [targetKey]
+  )
```

Comment:
Maybe simplify the repeated handler logic?:

```typescript
  const keyHandler = useCallback(
    (type: "up" | "down") => ({ key }: KeyboardEvent) => {
      if (key === targetKey) {
        setKeyPressed(type === "down")
      }
    },
    [targetKey]
  )

... 
// inside useEffect
const downHandler = keyHandler("down")
const upHandler = keyHandler("up")
```

Reply:
Refactored.